lib/collection/src/telemetry.rs

```diff
@@ -179,6 +177,12 @@ impl CollectionTelemetry {
             .filter(|i| i.replicate_states.values().any(|state| !state.is_active()))
             .count()
     }
+
+    pub fn optimization_time_spent(&self) -> usize {
```

Comment:
Style nit: either include time units in the name, or use proper type.
Also, please use `u64` instead of `usize` here and in other places.

```rust
pub fn optimization_time_spent_seconds(&self) -> u64 {
// OR
pub fn optimization_time_spent(&self) -> Duration {
// OR
pub fn optimization_time_spent(&self) -> TimeDelta {
```